core/src/test/java/io/substrait/extension/TypeExtensionTest.java

```diff
@@ -76,4 +77,30 @@ void roundtripCustomType() {
     var planReturned = protoPlanConverter.from(protoPlan);
     assertEquals(plan, planReturned);
   }
+
+  @Test
+  void arrayIndexScalarFunction() {
```

Comment:
minor: what do you think naming this something `roundtripNumberedAnyTypes`, which I think better describes what we're checking (i.e. that substrait-java can read in types like `any1` and `any1?`).

Reply:
Resolved. Thanks for the suggestion.